plugins/modules/net_tools/nmcli.py

```diff
@@ -1573,6 +1687,10 @@ def _compare_conn_params(self, conn_info, options):
                     value = value.upper()
                     # ensure current_value is also converted to uppercase in case nmcli changes behaviour
                     current_value = current_value.upper()
+                if key == 'gsm.apn' and current_value[0] == '"':
+                    # Depending on version nmcli adds double-qoutes to gsm.apn
+                    # Need to strip them in order to compare both
+                    current_value = current_value[1:-1]
```

Comment:
I am tempted to suggest using `current_value.strip('"')` instead, but I'm not sure how `nmcli` handles escaped `"` characters so this is probably okay as is.

Reply:
I spent some time to find out, when these double-qoutes were added to nmcli code. 
My Debian 10 (nmcli v1.14.6) does not do that, but my ArchLinux client (v1.32.10) does.

As I did not find out if, how, where and when these qoutes are insterted, I took with the savest (and ugliest) version I came up with.

Reply:
I did a quick test and `nmcli` strictly enforces the allowed characters for APN's:
```
Error: Failed to add 'gsm-gsm_test' connection: gsm.apn: '"internet.telekom"' contains invalid char(s) (use [A-Za-z._-])
```
So any additional double quotes would be formatting (I tested on RHEL 8.4 `nmcli tool, version 1.30.0-10.el8_4` and no double-quotes were added).
Therefore `strip('"')` is the way to go and the `current_value[0] == '"'` check can be removed as well.